### PR TITLE
Fix column types

### PR DESCRIPTION
**Pull request by amanmahajan7**
I will remove `isFrozen` in a follow up PR

### CODE REVIEW

File: packages/react-data-grid/src/ColumnMetrics.ts [hunk 1]
```diff
@@ -38,17 +40,17 @@ function setDefferedColumnWidths(columns: Column[], unallocatedWidth: number, mi
 
 function setColumnOffsets(columns: Column[]): void {
   let left = 0;
-  for (const column of columns) {
+  for (const column of columns as CalculatedColumn[]) {
```
Comment (amanmahajan7) [inline packages/react-data-grid/src/ColumnMetrics.ts]: We can probably modify this method and return a list of calculated columns instead. wdyt?

File: packages/react-data-grid/src/ColumnMetrics.ts [hunk 2]
```diff
@@ -63,11 +65,11 @@ export function recalculate(metrics: Metrics) {
 
   const frozenColumns = columns.filter(c => isFrozen(c));
   const nonFrozenColumns = columns.filter(c => !isFrozen(c));
-  columns = frozenColumns.concat(nonFrozenColumns);
-  columns.forEach((c, i) => c.idx = i);
+  const calculatedColumns = frozenColumns.concat(nonFrozenColumns) as CalculatedColumn[];
```
Comment (amanmahajan7) [inline packages/react-data-grid/src/ColumnMetrics.ts]: Then we won't need type casting

File: packages/react-data-grid/src/ColumnMetrics.ts (hunk 1 above)
Comment (MayhemYDG) [reply packages/react-data-grid/src/ColumnMetrics.ts]: Yeah, maybe we can merge some together so we can do
```ts
function x(columns: Column[]): CalculatedColumn[] {
  const cc: CalculatedColumn[] = [];

  for (const column of columns) {
    const left = ...
    const width = ...
    cc.push({ ...column, left, width });
  }

  return cc;
}
```

File: packages/react-data-grid/src/HeaderRow.tsx [hunk 3]
```diff
@@ -131,11 +130,11 @@ export default class HeaderRow extends React.Component<HeaderRowProps> {
   }
 
   setScrollLeft(scrollLeft: number): void {
-    this.props.columns.forEach((column?: Column) => {
-      const { key } = column!;
+    this.props.columns.forEach(column => {
```
Comment (MayhemYDG) [inline packages/react-data-grid/src/HeaderRow.tsx]: We can use for...of here as well.

File: packages/react-data-grid/src/Row.tsx [hunk 4]
```diff
@@ -66,10 +66,9 @@ export default class Row extends React.Component<RowRendererProps> implements Ro
 
   getCells() {
     const { colOverscanStartIdx, colOverscanEndIdx, columns } = this.props;
-    const frozenColumns = columns.filter((c?: Column) => isFrozen(c!)) as Column[];
-    const nonFrozenColumn = columns.slice(colOverscanStartIdx, colOverscanEndIdx + 1).filter((c?: Column) => !isFrozen(c!)) as Column[];
-    return nonFrozenColumn.concat(frozenColumns)
-      .map((c?: Column) => this.getCell(c!));
+    const frozenColumns = columns.filter(c => isFrozen(c));
```
Comment (MayhemYDG) [inline packages/react-data-grid/src/Row.tsx]:
```suggestion
    const frozenColumns = columns.filter(isFrozen);
```

**Pull request merged**